plugins/inventory/foreman.py

```diff
@@ -256,8 +256,20 @@ def _get_json(self, url, ignore_errors=None, params=None):
                     break
                 elif isinstance(json['results'], MutableMapping):
                     # /facts are returned as dict in 'results'
-                    results = json['results']
-                    break
+                    if not isinstance(results, MutableMapping):
+                        results = {}
+
+                    # check for end of paging
+                    if len(json['results']) == 0:
+                        break
+
+                    for host, facts in json['results'].items():
+                        if host not in results:
+                            results[host] = {}
```

Comment:
This could be prevented by using `results = defaultdict(dict)`.

Reply:
Sure, but I'm lazy :D